allolib-wasm/include/al_WebAudioVisualBridge.hpp

```
@@ -0,0 +1,392 @@
+    float beatDetected;          // 1.0 if beat, 0.0 otherwise
+    float beatEnergy;
+    float beatThreshold;
+    float tempo;                 // estimated BPM (0 if unknown)
+    float _pad;
+};
+static_assert(sizeof(AudioFeatures) == 48, "AudioFeatures must be 48 bytes");
+
+// ── Feature Extraction WGSL ─────────────────────────────────────────────────
+
+// Single workgroup of 256 threads, parallel reduction over FFT bins
+static const char* kFeatureExtractionWGSL = R"(
```

Clang (14.0.6)

[warning] variable 'kFeatureExtractionWGSL' is non-const and globally accessible, consider making it const

(cppcoreguidelines-avoid-non-const-global-variables)